### PR TITLE
Batch GEMM with alpha==0 and batch>1

### PR DESCRIPTION
**Pull request by ProGTX**
When `alpha == 0`, GEMM can just scale the matrix.
However, this currently doesn't take the `batch_size` into account,
leading to wrong results.

This PR fixes this issue and adds tests for the new cases.
It also updates existing tests by changing `alpha` and `beta`
to something other than `1` -
the issue this PR is fixing doesn't present itself when `beta == 1`.

### CODE REVIEW

File: test/unittest/blas3/blas3_gemm_batched_test.cpp [hunk 1]
```diff
@@ -34,8 +34,8 @@ const auto BetaNonZeroLDMatch =
                        ::testing::Values(63, 128),   // k
                        ::testing::Values('n', 't'),  // transa
                        ::testing::Values('n', 't'),  // transb
-                       ::testing::Values(1.0),       // alpha
-                       ::testing::Values(1.0),       // beta
+                       ::testing::Values(3.0),       // alpha
+                       ::testing::Values(7.0),       // beta
```
Comment (Rbiessy) [inline test/unittest/blas3/blas3_gemm_batched_test.cpp:38]: Why this change? There is no special case for alpha=1 or for beta=1 as far as I know.

File: test/unittest/blas3/blas3_gemm_batched_test.cpp [hunk 2]
```diff
@@ -50,11 +50,42 @@ const auto BetaNonZeroLDMultiplied =
...
```
Comment (mehdi-goli) [inline test/unittest/blas3/blas3_gemm_batched_test.cpp:54]: same here

File: test/unittest/blas3/blas3_gemm_batched_test.cpp (hunk 1 above)
Comment (ProGTX) [reply test/unittest/blas3/blas3_gemm_batched_test.cpp:38]: The issue this PR is fixing doesn't get triggered when `beta == 1`. I discovered that because I copy-pasted the original tests :) I thought it might be better in the future to have more complicated values to potentially capture more issues. I can remove it if you don't think that should be included in this PR.

Comment (Rbiessy) [reply test/unittest/blas3/blas3_gemm_batched_test.cpp:38]: Right, I don't mind if it stays.
